C/services/core/configuration_manager.cpp

```diff
@@ -123,50 +151,82 @@ ConfigCategory ConfigurationManager::getCategoryAllItems(const string& categoryN
 	Where *wKey = new Where("key", conditionKey, categoryName);
 	Query qKey(wKey);
 
-	// Query via storage client
-	ResultSet* categoryItems = m_storage->queryTable("configuration", qKey);
-	if (!categoryItems)
-	{
-		throw StorageOperationException();
-	}
-
-	// Cayegory not found
-	if (!categoryItems->rowCount())
+	ResultSet* categoryItems = 0;
+	try
 	{
-		delete categoryItems;
-		throw NoSuchCategoryException();
-	}
+		// Query via storage client
+		categoryItems = m_storage->queryTable("configuration", qKey);
+		if (!categoryItems)
+		{
+			throw ConfigCategoryEx();
+		}
 
-	// Get first row
-	ResultSet::RowIterator it = categoryItems->firstRow();
-	ResultSet::Row* row = *it;
-	if (!row)
-	{
-		delete categoryItems;
-		throw CategoryDetailsException();
-	}	
+		// Cayegory not found
+		if (!categoryItems->rowCount())
```

Comment:
Again you could combine these two tests

Reply:
The code here returns two different exceptions and we cannot combine it